content/en/apps/guides/hosting/4.x/data-migration.md

```diff
@@ -0,0 +1,244 @@
+export CHT_NETWORK=<docker-network-name>
+export COUCH_URL=http://admin:pass@docker-service-name:5984
+```
+
+For simplicity, you could store these required values in an `.env` file: 
+```shell
+cat > ${HOME}/couchdb-migration/.env << EOF
+CHT_NETWORK=cht-net
+COUCH_URL=http://admin:pass@127.0.0.1:5984
+EOF
+```
+
+### 2. Prepare CHT-Core 3.x installation for upgrading
+To minimize downtime when upgrading, it's advised to prepare the 3.x installation for the 4.x upgrade, and pre-index all views that are required by 4.x.
+
+The migration tool provides a command which will download all 4.x views to your 3.x CouchDb, and initiate view indexing.
+
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration pre-index-views <desired CHT version>
+```
+
+Once view indexing is finished, proceed with the next step.
+
+{{% alert title="Note" %}} If this step is omitted, 4.x API will fail to respond to requests until all views are indexed. Depending on the size of the database, this could take many hours, or even days. {{% /alert %}} 
+
+### 3. Save existent CouchDb configuration
+
+Some CouchDb configuration values must be ported from existent CouchDb to the 4.x installation. Store them in a safe location before shutting down 3.x CouchDb.
+
+##### a. CouchDB secret 
+Used in encrypting all CouchDb passwords and session tokens.
+##### b. CouchDb server uuid
+Used in generating replication checkpointer documents, which track where replication progress between every client and the server, and ensure that clients don't re-download or re-upload documents.
+
+Use the migration tool to obtain these values.
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration get-env
+```
+
+### 4. Locate and backup CouchDb Data folder
+a) If running in MedicOS, [CouchDb data folder]({{< relref "apps/guides/hosting/3.x/self-hosting#backup" >}}) can be found at `/srv/storage/medic-core/couchdb/data`.
+
+b) If running a custom installation of CouchDb, data would be typically stored at `/opt/couchdb/data`.
+
+c) TODO when using orchestration / AWS 
+
+### 5. Launch 4.x CouchDb installation
+
+Depending on your project scalability needs and technical possibilities, you must decide whether you will deploy CouchDb in a single node or in a cluster with multiple nodes.
+Please consult this guide about clustering and horizontal scalability to make an informed decision. <insert link>
+
+{{% alert title="Note" %}} You can start with single node and then change to a cluster. This involves running the migration tool again to distribute shards from the existent node to the new nodes. {{% /alert %}}
+
+Depending on your choice, follow the instructions that match your deployment below: 
+
+#### Single node
+
+a) Download 4.x single-node CouchDb docker-compose file:
+```shell
+mkdir -p ~/couchdb-single/ 
+cd ~/couchdb-single/ 
+curl -s -o ./docker-compose.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.0/docker-compose/cht-couchdb.yml
```

Comment:
I would trust that people will know which version of CHT CouchDb they want to install. Serving them with latest image might not be desirable.

Reply:
I can see not going latest - makes sense! 

I think we should make this be `<desired CHT version>` then so it matches the version specified in step 2 for `pre-index-views`.  Forcing them to choose so they get it right seems better instead of deciding for them and they might get it wrong.

Reply:
this still seems like a worthwhile change!

Reply:
Oh, so sorry I missed this!